### DISCUSSION TITLE
How can I add namespace info in metrics?

### DISCUSSION THREAD

**Discussion by csandanov** [distribution/distribution#4515] [Q&A]
I want to collect stats per namespace via prometheus

**Comment by milosgajdos** [comment]
I'd be careful about that, especially in large deployments where number of namespaces can be large so the label growth could be unbounded. 

If you are ok with that you could write a storage middleware (see [here](https://github.com/distribution/distribution/tree/e3007cd2bcd9526c1c7c92edad419bd0e67531d1/registry/storage/driver/middleware) for examples of existing middleware) which would add `prometheus.NewCounterVec` which could look something like this maybe (this is a quickly hacked up pseudocode provided with no warranty!):
```Go
var namespacePushCounter = prometheus.NewCounterVec(
    prometheus.CounterOpts{
        Name: "distribution_namespace_push_total",
        Help: "Counts the number of pushes to namespaces.",
    },
    []string{"namespace"},
)
```

Then in the storage middleware you would extract the namespace and increment the counter for it like so:
```Go
namespacePushCounter.With(prometheus.Labels{"namespace": SOME_NAMESPACE}).Inc()
```

In any case I would recommend to avoid doing this due to what I said above.

**Comment by csandanov** [comment]
Thanks! I will try it out
